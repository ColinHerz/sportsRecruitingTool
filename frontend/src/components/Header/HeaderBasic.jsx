import React, { Component } from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export default class extends Component {
  render() {
    var buttonName = this.props.buttonName;
    var buttonIcon = this.props.buttonIcon;
    var buttonLocation = this.props.buttonLocation;
    return (
      <header className="navbar" role="navigation" aria-label="navigation">
        <div className="navbar-menu">
          <div className="navbar-end">
            <span className="navbar-item">
              <Link to={buttonLocation}>
                <button className="button is-primary">
                  <span className="icon is-small is-left">
                    <FontAwesomeIcon icon={buttonIcon}></FontAwesomeIcon>
                  </span>
                  <span>{buttonName}</span>
                </button>
              </Link>
            </span>
          </div>
        </div>
      </header>
    );
  }
}
