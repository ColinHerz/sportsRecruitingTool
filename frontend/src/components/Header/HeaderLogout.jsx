import React, { Component } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export default class extends Component {

  logout = function(link) {
    localStorage.setItem("username", JSON.stringify(""));
    window.location.replace(link)
  }

  render() {
    var buttonName = this.props.buttonName;
    var buttonIcon = this.props.buttonIcon;
    var buttonLocation = this.props.buttonLocation;

    return (
      <header className="navbar" role="navigation" aria-label="navigation">
        <div className="navbar-menu">
          <div className="navbar-end">
            <span className="navbar-item">
              <button
                onClick={() => this.logout(buttonLocation)}
                className="button is-primary">
                <span className="icon is-small is-left">
                  <FontAwesomeIcon icon={buttonIcon}></FontAwesomeIcon>
                </span>
                <span>{buttonName}</span>
              </button>
            </span>
          </div>
        </div>
      </header>
    );
  }
}
