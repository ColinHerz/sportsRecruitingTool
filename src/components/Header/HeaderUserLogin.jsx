import React, { Component } from "react";
import axios from "axios";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export default class extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };
    this.removeDanger = this.removeDanger.bind(this);
    this.onChangeUsername = this.onChangeUsername.bind(this);
    this.onChangePassword = this.onChangePassword.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChangeUsername(e) {
    this.setState({
      username: e.target.value
    });
  }

  onChangePassword(e) {
    this.setState({
      password: e.target.value
    });
  }

  removeDanger() {
    if (document.getElementById("username").classList.contains("is-danger")) {
      document.getElementById("username").classList.remove("is-danger");
      document.getElementById("invalidUsername").style.display = "none";
      document.getElementById("password").classList.remove("is-danger");
      document.getElementById("invalidPassword").style.display = "none";
    }
  }

  onSubmit(e) {
    e.preventDefault();

    const user = {
      username: this.state.username,
      password: this.state.password
    };

    axios
      .post("/users/login", user)
      .then(res => {
        console.log(res.data.username);
        localStorage.setItem("username", JSON.stringify(this.state.username));
        window.location.href = "/contacts";
      })
      .catch(err => {
        if (err.response.status === 301) {
          document.getElementById("username").classList.add("is-danger");
          document.getElementById("password").classList.add("is-danger");
          document.getElementById("invalidUsername").style.display = "block";
          document.getElementById("invalidPassword").style.display = "block";
        }
      });
  }

  render() {
    return (
      <header
        className="navbar is-pulled-right"
        role="navigation"
        aria-label="navigation"
      >
        <div className="container">
          <form onSubmit={this.onSubmit}>
            <div className="navbar-menu">
              <div className="navbar-end">
                <span className="navbar-item">
                  <div className="control has-icons-left has-icons-right">
                    <input
                      className="input is-primary"
                      id="username"
                      required
                      value={this.state.username}
                      onChange={this.onChangeUsername}
                      type="text"
                      autoComplete="username"
                      placeholder="Username"
                      onFocus={this.removeDanger}
                    ></input>
                    <span className="icon is-small is-left">
                      <FontAwesomeIcon icon="user"></FontAwesomeIcon>
                    </span>
                    <div hidden id="invalidUsername">
                      <span className="icon is-small is-right">
                        <FontAwesomeIcon icon="exclamation-triangle"></FontAwesomeIcon>
                      </span>
                    </div>
                  </div>
                </span>
                <span className="navbar-item">
                  <div className="control has-icons-left has-icons-right">
                    <input
                      className="input is-primary"
                      id="password"
                      required
                      value={this.state.password}
                      onChange={this.onChangePassword}
                      type="password"
                      autoComplete="new-password"
                      placeholder="Password"
                      onFocus={this.removeDanger}
                    ></input>
                    <span className="icon is-small is-left">
                      <FontAwesomeIcon icon="lock"></FontAwesomeIcon>
                    </span>
                    <div hidden id="invalidPassword">
                      <span className="icon is-small is-right">
                        <FontAwesomeIcon icon="exclamation-triangle"></FontAwesomeIcon>
                      </span>
                    </div>
                  </div>
                </span>
                <span className="navbar-item">
                  <button type="submit" className="button is-primary">
                    <span className="icon is-small is-left">
                      <FontAwesomeIcon icon="sign-in-alt"></FontAwesomeIcon>
                    </span>
                    <span>Login</span>
                  </button>
                </span>
              </div>
            </div>
          </form>
        </div>
      </header>
    );
  }
}
