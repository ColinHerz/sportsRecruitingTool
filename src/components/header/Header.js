import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";
import React from "react";

import "./header.scss";

const Header = props => {
	return (
		<header>
			<NavLink to="/" id="logo"><h1>Sporta</h1></NavLink>

			{
				// profile will need to change to logged in user
				props.loggedIn ?
					<nav>
						<NavLink to="/events/">Events</NavLink>

						<NavLink to="/scores/">Top Scores</NavLink>

						<NavLink to="/profile/garbage/">My Profile</NavLink>
					</nav>:
					<div>
						<button
							id="log-in"
							onClick={props.logIn}
						>
						Log In
						</button>

						<button
							id="sign-up"
							onClick={props.register}
						>
						Sign Up
						</button>
					</div>
			}
		</header>
	);
};

Header.propTypes = {
	logIn: PropTypes.func.isRequired,
	register: PropTypes.func.isRequired,
	loggedIn: PropTypes.bool.isRequired
};

export default Header;
