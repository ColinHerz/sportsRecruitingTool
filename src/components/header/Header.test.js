/* eslint-env jest */

// MemoryRouter needed because header contains NavLinks, which need to
// be rendered within a Router
import { MemoryRouter } from "react-router-dom";
import { mount } from "../../../enzyme.js";
import React from "react";
import renderer from "react-test-renderer";

import Header from "./Header.js";

const genProps = loggedIn => {
	return {
		loggingIn: jest.fn(),
		registering: jest.fn(),
		loggedIn: loggedIn,
		logOut: jest.fn()
	};
};

beforeEach(() => {
	jest.resetModules();
});

describe(`Rendering`, () => {
	it(`renders correctly when not logged in`, () => {
		const props = genProps(false);

		const component = renderer.create(
			<MemoryRouter>
				<Header {... props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it(`renders correctly when logged in`, () => {
		const props = genProps(true);

		const component = renderer.create(
			<MemoryRouter>
				<Header {... props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe(`Functionality`, () => {
	it(`calls logging in when log in button clicked`, () => {
		const props = genProps(false);

		const component = mount(
			<MemoryRouter>
				<Header {... props} />
			</MemoryRouter>
		);

		component.find(`#log-in`).simulate(`click`);

		expect(props.loggingIn).toHaveBeenCalled();

		component.unmount();
	});

	it(`calls registering in when register button clicked`, () => {
		const props = genProps(false);

		const component = mount(
			<MemoryRouter>
				<Header {... props} />
			</MemoryRouter>
		);

		component.find(`#sign-up`).simulate(`click`);

		expect(props.registering).toHaveBeenCalled();

		component.unmount();
	});
});
