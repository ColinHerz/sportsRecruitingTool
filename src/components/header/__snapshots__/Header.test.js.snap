// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly when logged in 1`] = `
<header>
  <a
    aria-current="page"
    className="active"
    href="/"
    id="logo"
    onClick={[Function]}
    style={Object {}}
  >
    <h1>
      Sporta
    </h1>
  </a>
  <nav>
    <a
      aria-current={null}
      href="/myevents/garbage/"
      onClick={[Function]}
    >
      Events
    </a>
    <a
      aria-current={null}
      href="/scores/"
      onClick={[Function]}
    >
      Top Scores
    </a>
    <a
      aria-current={null}
      href="/profile/garbage/"
      onClick={[Function]}
    >
      My Profile
    </a>
    <button
      onClick={[MockFunction]}
    >
      Log Out
    </button>
  </nav>
</header>
`;

exports[`Rendering renders correctly when not logged in 1`] = `
<header>
  <a
    aria-current="page"
    className="active"
    href="/"
    id="logo"
    onClick={[Function]}
    style={Object {}}
  >
    <h1>
      Sporta
    </h1>
  </a>
  <div>
    <button
      id="log-in"
      onClick={[MockFunction]}
    >
      Log In
    </button>
    <button
      id="sign-up"
      onClick={[MockFunction]}
    >
      Sign Up
    </button>
  </div>
</header>
`;
